### ISSUE TITLE
AG Charts Vue does not work with composition API/script setup

### ISSUE THREAD

**Issue by sk-tjdownes** [ag-grid/ag-charts#699]
**I'm submitting a ...** 
feature request

**Providing a Reproducible Scenario**
[Repro Sandbox](https://codesandbox.io/p/sandbox/jovial-sky-hm4fxv?file=%2Fsrc%2FApp.vue)

**Current behavior**
Installed AG Charts Community, AG Charts Vue with Vue 3 on Vite.
Exception occurs at runtime, stating 
`Could not find dependency: 'vue-class-component' relative to '/node_modules/ag-charts-vue3/lib/AgChartsVue.js'`

**Expected behavior**
Since Vue maintainers have clearly stated that composition API and script setup are The Way, AG Grid Vue should support these by default.

At the very least, the documentation should state that it does not work with script setup at this time

Vue 3.10.10
Ag Charts Community 9.0.0
AG Charts Vue 3 9.0.0

-   **Browser:**
Any

-   **Language:** ES6/7

**Comment by AG-DavidG** [comment]
Hi @sk-tjdownes 

I see you've closed this issue yourself, but I just wanted to check that you were able to get this to work.

Please do let me know if there's anything we can improve in our library or documentation around our support for Vue.

Thanks

David

**Comment by jacobp100** [comment]
For now, you'll need to install `vue-class-component`, since our package currently lists this as a peer dependency

What would 'ideal' look like for you?

**Comment by sk-tjdownes** [comment]
> I see you've closed this issue yourself, but I just wanted to check that you were able to get this to work.

Yes, thanks. I realized I needed to install `vue-class-component`. Everything was good after that. Oddly, I thought I installed it initially and was getting odd errors, but it likely could have been a package name typo.
 
> Please do let me know if there's anything we can improve in our library or documentation around our support for Vue.

I think adding that `ag-charts-community` and `vue-class-component` as requirements in the docs would  simplify getting up and running. 

Also, `vue-class-component` is deprecated, and the repo suggests a replacement. It would be good to see this switched in the vue charts plugin.

Additionally it would be nice to have some details regarding testing. I ended up mocking canvas with `vitest-canvas-mock`, after some research.

Thanks @AG-DavidG anf @jacobp100
